grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -128,40 +120,41 @@ class StreamService {
 
 	// Ref to Kafka will be abstracted out when Feed abstraction is reworked
 
-	void sendMessage(Stream stream, String partitionKey, long timestamp, byte[] content, byte contentType, int ttl=0,
-					 SignatureType signatureType = SignatureType.SIGNATURE_TYPE_NONE, String address = null, String signature = null) {
+	void sendMessage(Stream stream, String partitionKey, long timestamp, String content, StreamMessage.ContentType contentType,
+					 StreamMessage.SignatureType signatureType = StreamMessage.SignatureType.SIGNATURE_TYPE_NONE, String address = "", String signature = null) {
 		int streamPartition = partitioner.partition(stream, partitionKey)
-		StreamrBinaryMessage msg = new StreamrBinaryMessageV29(stream.id, streamPartition, timestamp, ttl, contentType, content,
-			signatureType, address, signature)
+		StreamMessage msg = new StreamMessageV30(stream.id, streamPartition, timestamp,
+			0, address, (Long) null, 0, contentType, content, signatureType, signature)
 
 		String kafkaPartitionKey = "${stream.id}-$streamPartition"
 		kafkaService.sendMessage(msg, kafkaPartitionKey)
 	}
 
 	@CompileStatic
-	void sendMessage(Stream stream, Map message, int ttl=0) {
+	void sendMessage(Stream stream, Map message) {
 		String str = gson.toJson(message)
-		sendMessage(stream, null, System.currentTimeMillis(), str.getBytes(utf8), StreamrBinaryMessage.CONTENT_TYPE_JSON, ttl);
+		sendMessage(stream, null, System.currentTimeMillis(), str, StreamMessage.ContentType.CONTENT_TYPE_JSON)
 	}
 
 	@CompileStatic
-	void sendMessage(Stream stream, long timestamp, Map message, int ttl=0) {
+	void sendMessage(Stream stream, long timestamp, Map message) {
 		String str = gson.toJson(message)
-		sendMessage(stream, null, timestamp, str.getBytes(utf8), StreamrBinaryMessage.CONTENT_TYPE_JSON, ttl);
+		sendMessage(stream, null, timestamp, str, StreamMessage.ContentType.CONTENT_TYPE_JSON)
 	}
 
 	@CompileStatic
-	void sendMessage(Stream stream, @Nullable String partitionKey, Map message, int ttl=0) {
+	void sendMessage(Stream stream, @Nullable String partitionKey, Map message) {
 		String str = gson.toJson(message)
-		sendMessage(stream, partitionKey, System.currentTimeMillis(), str.getBytes(utf8), StreamrBinaryMessage.CONTENT_TYPE_JSON, ttl);
+		sendMessage(stream, partitionKey, System.currentTimeMillis(), str, StreamMessage.ContentType.CONTENT_TYPE_JSON)
 	}
 
-	void saveMessage(Stream stream, String partitionKey, long timestamp, Map message, int ttl, long messageNumber, Long previousMessageNumber) {
-		int streamPartition = partitioner.partition(stream, partitionKey)
+	StreamMessage saveMessage(Stream stream, int streamPartition, long timestamp, long sequenceNumber, String publisherId, Map message, long previousTimestamp, long previousSequenceNumber) {
```

Comment:
If we expose just `sendMessage(StreamMessage)`, then here too `saveMessage(StreamMessage)` becomes the new standard!